### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by SerhiiUnhurian**
- [DEMO LINK](https://SerhiiUnhurian.github.io/layout_calendar/)
- [TEST REPORT LINK](https://SerhiiUnhurian.github.io/layout_calendar/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -7,6 +7,38 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Calendar</h1>
+    <div class="calendar calendar--start-day--sun calendar--month-length--28">
+      <div class="calendar__sell calendar__sell--1"></div>
```
Comment (denys-danyliuk) [inline src/index.html]:
```suggestion
      <div class="calendar__cell"></div>
```
you don't need to add modifiers on cells, only on calendar

File: src/styles/calendar.scss [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    transition: all 0.5s;
+
+    @for $day from 1 through 31 {
+      &--#{$day} {
+        &::before {
+          content: "#{$day}";
+          position: absolute;
+          font-size: 30px;
+          font-family: Arial, Helvetica, sans-serif;
+        }
```
Comment (denys-danyliuk) [inline src/styles/calendar.scss]: 7'th point in checklist
not all of this styles changing on each iteration

File: src/styles/calendar.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
+.calendar {
+  padding: 20px 10px;
+  width: 7 * ($sell-size + $sell-gap);
+  border: 2px solid black;
```
Comment (denys-danyliuk) [inline src/styles/calendar.scss]: remove this border

File: src/styles/calendar.scss [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    box-sizing: border-box;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: $sell-size;
+    height: $sell-size;
+    border: 1px solid black;
+    position: relative;
+    background-color: $Sell-color;
+    transition: all 0.5s;
```
Comment (denys-danyliuk) [inline src/styles/calendar.scss]: list styles that you apply transition to instead of "all"

**Review by denys-danyliuk — CHANGES_REQUESTED** [review]

File: src/styles/calendar.scss (hunk 4 above)
Comment (SerhiiUnhurian) [reply src/styles/calendar.scss]: okay

File: src/styles/calendar.scss (hunk 3 above)
Comment (SerhiiUnhurian) [reply src/styles/calendar.scss]: done

**Comment by SerhiiUnhurian** [comment]
i'll remove the sell modifier, after the review, for now i just commented it out

File: src/styles/calendar.scss [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    //   &--#{$day} {
+    //     &::before {
+    //       content: "#{$day}";
+    //       position: absolute;
+    //     }
+    //   }
+    // }
+
+    &:hover {
+      cursor: pointer;
```
Comment (Esceype) [inline src/styles/calendar.scss]: This can be in element above

File: src/styles/calendar.scss [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  @for $month-length from 28 through 31 {
+    &--month-length--#{$month-length} {
+      .calendar__sell:nth-child(n + #{$month-length + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__sell {
```
Comment (Esceype) [inline src/styles/calendar.scss]: Rename this element name cause at the moment this naming can misinform

File: src/index.html (hunk 1 above)
Comment (Esceype) [reply src/index.html]: Remove this modifier if you don`t need it now

File: src/styles/calendar.scss [hunk 7]
```diff
@@ -0,0 +1,58 @@
+.calendar {
+  padding: 20px 10px;
+  width: 7 * ($sell-size + $sell-gap);
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
```
Comment (Esceype) [inline src/styles/calendar.scss:6]: Do not use gap, tests don`t expect it

**Review by Esceype — CHANGES_REQUESTED** [review]
At all great job just remove the modifier at each day, cause now it can work without them

File: src/styles/calendar.scss [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  flex-wrap: wrap;
+
+  & :not(:nth-child(7n)) {
+    margin-right: 1px;
+  }
+
+  @for $day from 1 through 31 {
+    & > &__day:nth-child(#{$day}):before {
+      content: "#{$day}";
+      position: absolute;
```
Comment (denys-danyliuk) [inline src/styles/calendar.scss]: does this property on each iteration?

File: src/styles/calendar.scss [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    box-sizing: border-box;
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+    @include flexCenter;
+    width: $sell-size;
+    height: $sell-size;
+    border: 1px solid black;
+    position: relative;
+    background-color: $sell-color;
+    transition: transform 0.5s;
```
Comment (denys-danyliuk) [inline src/styles/calendar.scss]: 8'th point in checklist

File: src/styles/calendar.scss (hunk 8 above)
Comment (denys-danyliuk) [reply src/styles/calendar.scss]: 7'th point in checklist

**Review by denys-danyliuk — CHANGES_REQUESTED** [review]

File: src/styles/calendar.scss [hunk 10]
```diff
@@ -0,0 +1,52 @@
+.calendar {
+  padding: 20px 10px;
+  width: 7 * ($sell-size + $sell-gap);
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+
+  // & :not(:nth-child(7n)) {
+  //   margin: 1px;
+  // }
```
Comment (denys-danyliuk) [inline src/styles/calendar.scss:10]: delete comments

**Review by denys-danyliuk — APPROVED** [review]

**Pull request closed without merging**